scrunch/mutable_dataset.py

```diff
@@ -0,0 +1,892 @@
+            then it will be preserved regardless of this parameter.
+
+        :returns _fork: scrunch.datasets.Dataset
+            The forked dataset.
+        """
+
+        nforks = len(self.resource.forks.index)
+        if name is None:
+            name = "FORK #{} of {}".format(nforks + 1, self.resource.body.name)
+        if description is None:
+            description = self.resource.body.description
+
+        body = dict(
+            name=name,
+            description=description,
+            is_published=is_published,
+            **kwargs
+        )
+        # not returning a dataset
+        _fork = self.resource.forks.create({"body": body}).refresh()
+        fork = MutableDataset(_fork)
+
+        if preserve_owner or '/api/projects/' in self.resource.body.owner:
+            try:
+                _fork.patch({'owner': self.resource.body.owner})
+                _fork.refresh()
+            except (pycrunch.ClientError, pycrunch.ServerError):
+                # Don't fail if the fork's owner cannot be set to be the same
+                # as the parent dataset.
+                pass
+
+        return fork
+
+    def create_multitable(self, name, template, is_public=False):
+        """
+        template: List of dictionaries with the following keys
+        {"query": <query>, "transform": <transform>|optional}.
+        A query is a variable or a function on a variable:
+        {"query": bin(birthyr)}
+
+        If transform is specified it must have the form
+        {
+            "query": var_x,
+            "transform": {"categories": [
+                {
+                    "missing": false,  --> default: False
+                    "hide": true,      --> default: True
+                    "id": 1,
+                    "name": "not asked"
+                },
+                {
+                    "missing": false,
+                    "hide": true,
+                    "id": 4,
+                    "name": "skipped"
+                }
+            ]}
+        }
+        """
+        # build template payload
+        parsed_template = []
+        for q in template:
+            # sometimes q is not a dict but simply a string, convert it to a dict
+            if isinstance(q, str):
+                q = {"query": q}
+            as_json = {}
+            parsed_q = process_expr(parse_expr(q['query']), self.resource)
+            # wrap the query in a list of one dict element
+            as_json['query'] = [parsed_q]
+            if 'transform' in q.keys():
+                as_json['transform'] = q['transform']
+            parsed_template.append(as_json)
+        payload = dict(element='shoji:entity',
+                       body=dict(name=name,
+                                 is_public=is_public,
+                                 template=parsed_template))
+        new_multi = self.resource.multitables.create(payload)
+        return self.multitables[new_multi.body['name']]
```

Comment:
but I think it's really just a dict of tuples and not an actual Entity, and I don't think a DS would have thousands of such subentities.